packages/console/src/pages/Applications/components/Guide/index.tsx

```diff
@@ -61,68 +62,70 @@ function Guide({ className, guideId, app, isCompact, onClose }: Props) {
       {!isCompact && <GuideHeader onClose={onClose} />}
       <div className={styles.content}>
-        <GuideContext.Provider value={memorizedContext}>
-          <MDXProvider
-            components={{
-              code: ({ className, children }) => {
-                const [, language] = /language-(\w+)/.exec(String(className ?? '')) ?? [];
+        {memorizedContext ? (
+          <GuideContext.Provider value={memorizedContext}>
+            <MDXProvider
+              components={{
+                code: ({ className, children }) => {
+                  const [, language] = /language-(\w+)/.exec(String(className ?? '')) ?? [];
 
-                return language ? (
-                  <CodeEditor
-                    isReadonly
-                    // We need to transform `ts` to `typescript` for prismjs, and
-                    // it's weird since it worked in the original Guide component.
-                    // To be investigated.
-                    language={language === 'ts' ? 'typescript' : language}
-                    value={String(children).trimEnd()}
-                  />
-                ) : (
-                  <code>{String(children).trimEnd()}</code>
-                );
-              },
-              a: ({ children, ...props }) => (
-                <TextLink {...props} target="_blank" rel="noopener noreferrer">
-                  {children}
-                </TextLink>
-              ),
-              details: DetailsSummary,
-            }}
-          >
-            <Suspense fallback={<StepsSkeleton />}>
-              {tenantEndpoint && <GuideComponent {...memorizedContext} />}
-            </Suspense>
-          </MDXProvider>
-        </GuideContext.Provider>
-        {!isCompact && (
+                  return language ? (
+                    <CodeEditor
+                      isReadonly
+                      // We need to transform `ts` to `typescript` for prismjs, and
+                      // it's weird since it worked in the original Guide component.
+                      // To be investigated.
+                      language={language === 'ts' ? 'typescript' : language}
+                      value={String(children).trimEnd()}
+                    />
+                  ) : (
+                    <code>{String(children).trimEnd()}</code>
+                  );
+                },
+                a: ({ children, ...props }) => (
+                  <TextLink {...props} target="_blank" rel="noopener noreferrer">
+                    {children}
+                  </TextLink>
+                ),
+                details: DetailsSummary,
+              }}
+            >
+              <Suspense fallback={<StepsSkeleton />}>
+                {tenantEndpoint && GuideComponent && <GuideComponent {...memorizedContext} />}
+              </Suspense>
+            </MDXProvider>
+          </GuideContext.Provider>
+        ) : (
+          <NotFound className={styles.notFound} />
```

Comment:
nit: can we throw an error (sth like 404 request error) and leave it to the outer error boundary?

Reply:
No, I think it would be unnecessary, since this only happens when someone input an invalid URL on purpose, exactly like manually input any other invalid router URL. And the result should be the same, that we show this 404 image